test/metabase/events/notification_test.clj

```diff
@@ -0,0 +1,66 @@
+(ns metabase.events.notification-test
+  (:require
+   [clojure.test :refer :all]
+   [metabase.events :as events]
+   [metabase.events.notification :as events.notification]
+   [metabase.models.notification :as models.notification]
+   [metabase.notification.core :as notification]
+   [metabase.test :as mt]))
+
+(def supported-topics @#'events.notification/supported-topics)
+
+(defmacro with-temporary-event-topics
+  "Temporarily make `topics` valid event topics."
+  [topics & body]
+  `(let [topics# ~topics]
+     (try
+       (doseq [topic# topics#]
+         (derive topic# :metabase/event))
+       ~@body
+       (finally
+         (doseq [topic# topics#]
+           (underive topic# :metabase/event))))))
+
+(deftest supported-events-with-notification-will-be-sent-test
+  (mt/with-model-cleanup [:model/Notification]
+    (with-temporary-event-topics #{:event/test-notification}
+      (let [topic      :event/test-notification
+            noti-1     (models.notification/create-notification!
+                        {:payload_type :notification/system-event
+                         :active       true}
+                        [{:type       :notification-subscription/system-event
+                          :event_name topic}])
+            noti-2     (models.notification/create-notification!
+                        {:payload_type :notification/system-event
+                         :active       true}
+                        [{:type       :notification-subscription/system-event
+                          :event_name topic}])
+            _inactive  (models.notification/create-notification!
+                        {:payload_type :notification/system-event
+                         :active       false}
+                        [{:type       :notification-subscription/system-event
+                          :event_name topic}])
+            sent-notis (atom #{})]
```

Comment:
I think its worth testing that we're not sending duplicates, i.e. make this hold an array, and use `sort` to make the assertion order independent.